scripts/utils/internet_detect_tool.py

```diff
@@ -0,0 +1,246 @@
+from PyQt5.QtWidgets import QApplication, QMenu
+from PyQt5.QtCore import QTimer, QThread, pyqtSignal
+import threading
+import requests
+import urllib3
+import win32api
+import win32gui
+import win32con
+import winreg
+import struct
+import signal
+import ctypes
+import time
+import os
+
+
+# Define constants
+CHECK_INTERVAL = 2  # Seconds
+CONNECT_TEST_URL = "https://www.msftconnecttest.com/connecttest.txt"
```

Comment:
I think we should have a hash with TEST_URLs and their EXPECTED_RESPONSE, as if this concrete URL is down, the internet test would fail.

Reply:
what about testing the adapter in addition?